hedera-node/src/test/java/com/hedera/services/txns/schedule/ScheduleCreateTransitionLogicTest.java

```diff
@@ -69,6 +86,81 @@ public void hasCorrectApplicability() {
         assertFalse(subject.applicability().test(TransactionBody.getDefaultInstance()));
     }
 
+    @Test
+    public void followsHappyPath() {
+        // given:
+        givenValidTxnCtx();
+
+        // and:
+        given(store.getScheduleIDByTransactionBody(any())).willReturn(EMPTY_SCHEDULE);
+        given(store.createProvisionally(any(), any(), any(), any(), any())).willReturn(scheduleCreationResult);
+        given(store.addSigners(any(), any())).willReturn(OK);
+
+        // when:
+        subject.doStateTransition();
+
+        // then:
+        verify(store).createProvisionally(any(), any(), any(), any(), any());
+        verify(store).addSigners(any(), any());
+        verify(store).commitCreation();
+        verify(txnCtx).setStatus(SUCCESS);
+    }
+
+    @Test
+    public void capturesPendingScheduledTransaction() {
+        // given:
+        givenValidTxnCtx();
+
+        // and:
+        given(store.getScheduleIDByTransactionBody(any())).willReturn(Optional.of(schedule));
+
+        // when:
+        subject.doStateTransition();
+
+        // then:
+        verify(store, never()).createProvisionally(any(), any(), any(), any(), any());
+        verify(store).addSigners(eq(schedule), any());
+        verify(store, never()).commitCreation();
+        verify(txnCtx, never()).setStatus(SUCCESS);
+    }
+
+    @Test
+    public void capturesFailingCreateProvisionally() {
+        // given:
+        givenValidTxnCtx();
+
+        // and:
+        given(store.getScheduleIDByTransactionBody(any())).willReturn(EMPTY_SCHEDULE);
+        given(store.createProvisionally(any(), any(), any(), any(), any())).willReturn(EMPTY_CREATION_RESULT);
+
+        subject.doStateTransition();
+
+        // then:
+        verify(store).createProvisionally(any(), any(), any(), any(), any());
+        verify(store, never()).addSigners(any(), any());
+        verify(store, never()).commitCreation();
+        verify(txnCtx, never()).setStatus(SUCCESS);
+    }
+
+    @Test
+    public void capturesFailingSignersAddition() {
+        // given:
+        givenValidTxnCtx();
+
+        // and:
+        given(store.getScheduleIDByTransactionBody(any())).willReturn(EMPTY_SCHEDULE);
+        given(store.createProvisionally(any(), any(), any(), any(), any())).willReturn(CreationResult.success(schedule));
+        given(store.addSigners(any(), any())).willReturn(NOT_OK);
+
+        subject.doStateTransition();
+
+        // then:
+        verify(store).createProvisionally(any(), any(), any(), any(), any());
+        verify(store).addSigners(any(), any());
```

Comment:
I think that we should very the exact parameters to be sure that methods were properly called and not just called. F.e we won't be sure that the signature pair list is iterated and the proper keys are added if we do not verify the parameters of the method